### PR TITLE
rpc: verify dcrwallet's have --manualtickets set.

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: rpc/dcrwallet.go [hunk 1]
```diff
@@ -121,6 +121,13 @@ func (w *WalletConnect) Clients(ctx context.Context, netParams *chaincfg.Params)
 			continue
 		}
 
+		if !walletInfo.ManualTickets {
+			// All wallet's should not be adding tickets found via the network.
+			log.Errorf("wallet does not have manual tickets enabled (wallet=%s)", c.String())
+			failedConnections = append(failedConnections, connect.addr)
```
Comment (jholdstock) [inline rpc/dcrwallet.go:127]: I don't think this needs to be counted as a failed wallet connection. All of the wallet RPCs will still be available, and as we discussed on Matrix, misconfiguration here will not have any negative impact on VSP users, only (potentially) the service operator.

Should be fine to just log an error and proceed with using the connection - see below for `.Voting` and `.Unlocked` examples.

**Review by jholdstock — CHANGES_REQUESTED** [review]

**Comment by jholdstock** [comment]
FYI @isuldor

This should be the fix for dcrwallet voting on tickets which have the same voting address but haven't paid a fee. Requires decred/dcrwallet#1833 and decred/dcrwallet#1849

**Comment by isuldor** [comment]
Unless I'm just doing something wrong, ticketbuyer can't seem to make any purchases
```
2020-09-22 22:21:08.257 [INF] SYNC: Connected block 000000053c978ebec805c259f4d12dc9f784cb2cca7ddffcd2ead7d7f8f211ba, height 518725, 0 wallet transaction(s)
2020-09-22 22:21:08.913 [INF] TKBY: buy: 7, fee: 119627
2020-09-22 22:21:08.913 [INF] TKBY: 203fdd4d3016e2774f19c68613f49ae1839e52e99aa4315a24196f54a5e86959:2
2020-09-22 22:21:08.913 [INF] TKBY: 49f2ad4115441b40ee5e3d51435e569f75059da14d618d01d99d6692aac5b673:2
2020-09-22 22:21:08.913 [INF] TKBY: 909253c30d9eb326935075afc5c8347744b09eb50342211f003f9ccc43fc1f7c:2
2020-09-22 22:21:08.913 [INF] TKBY: b3d7b4b3dbe1e6ad30a66ecfac8b717159746928d57acd3ca8e9dc9ab022ebb0:2
2020-09-22 22:21:08.913 [INF] TKBY: 7a30df0c57f5c5be4f0ab9f40ec5025c606c1e261bd57d87af5346418fdfddb9:2
2020-09-22 22:21:08.913 [INF] TKBY: fad67191d32e7e4fd5858a568179caf8efd7a52e3a4ed2aa7ecbba8fdbaeb6cf:1
2020-09-22 22:21:08.913 [INF] TKBY: daf6de0381be55e4b0e1ebb12c44eef2a0c7e718ba4f9de65e57ce29ffc30ad3:2
2020-09-22 22:21:08.918 [INF] WLLT: Returning address (account=0 branch=1 child=5275)
2020-09-22 22:21:08.918 [INF] WLLT: Returning address (account=0 branch=1 child=5276)
2020-09-22 22:21:08.939 [ERR] TKBY: BUY: wallet.PurchaseTicketsWithResponse: insufficient balance:: txauthor.NewUnsignedTransaction
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit 203fdd4d3016e2774f19c68613f49ae1839e52e99aa4315a24196f54a5e86959:2
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit 49f2ad4115441b40ee5e3d51435e569f75059da14d618d01d99d6692aac5b673:2
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit 909253c30d9eb326935075afc5c8347744b09eb50342211f003f9ccc43fc1f7c:2
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit b3d7b4b3dbe1e6ad30a66ecfac8b717159746928d57acd3ca8e9dc9ab022ebb0:2
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit 7a30df0c57f5c5be4f0ab9f40ec5025c606c1e261bd57d87af5346418fdfddb9:2
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit fad67191d32e7e4fd5858a568179caf8efd7a52e3a4ed2aa7ecbba8fdbaeb6cf:1
2020-09-22 22:21:08.939 [INF] TKBY: unlocked unneeded credit daf6de0381be55e4b0e1ebb12c44eef2a0c7e718ba4f9de65e57ce29ffc30ad3:2
2020-09-22 22:25:25.144 [INF] SYNC: Connected block 000000421d624b168e26840b79ad67243c6207c17a1d7eb90d3bb379382c850f, height 518726, 0 wallet transaction(s)
2020-09-22 22:25:25.797 [INF] TKBY: buy: 7, fee: 119627
2020-09-22 22:25:25.797 [INF] TKBY: 203fdd4d3016e2774f19c68613f49ae1839e52e99aa4315a24196f54a5e86959:2
2020-09-22 22:25:25.797 [INF] TKBY: 49f2ad4115441b40ee5e3d51435e569f75059da14d618d01d99d6692aac5b673:2
2020-09-22 22:25:25.797 [INF] TKBY: 909253c30d9eb326935075afc5c8347744b09eb50342211f003f9ccc43fc1f7c:2
2020-09-22 22:25:25.797 [INF] TKBY: b3d7b4b3dbe1e6ad30a66ecfac8b717159746928d57acd3ca8e9dc9ab022ebb0:2
2020-09-22 22:25:25.797 [INF] TKBY: 7a30df0c57f5c5be4f0ab9f40ec5025c606c1e261bd57d87af5346418fdfddb9:2
2020-09-22 22:25:25.797 [INF] TKBY: fad67191d32e7e4fd5858a568179caf8efd7a52e3a4ed2aa7ecbba8fdbaeb6cf:1
2020-09-22 22:25:25.797 [INF] TKBY: daf6de0381be55e4b0e1ebb12c44eef2a0c7e718ba4f9de65e57ce29ffc30ad3:2
2020-09-22 22:25:25.812 [INF] WLLT: Returning address (account=0 branch=1 child=5277)
2020-09-22 22:25:25.812 [INF] WLLT: Returning address (account=0 branch=1 child=5278)
2020-09-22 22:25:25.827 [ERR] TKBY: BUY: wallet.PurchaseTicketsWithResponse: insufficient balance:: txauthor.NewUnsignedTransaction
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit 203fdd4d3016e2774f19c68613f49ae1839e52e99aa4315a24196f54a5e86959:2
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit 49f2ad4115441b40ee5e3d51435e569f75059da14d618d01d99d6692aac5b673:2
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit 909253c30d9eb326935075afc5c8347744b09eb50342211f003f9ccc43fc1f7c:2
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit b3d7b4b3dbe1e6ad30a66ecfac8b717159746928d57acd3ca8e9dc9ab022ebb0:2
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit 7a30df0c57f5c5be4f0ab9f40ec5025c606c1e261bd57d87af5346418fdfddb9:2
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit fad67191d32e7e4fd5858a568179caf8efd7a52e3a4ed2aa7ecbba8fdbaeb6cf:1
2020-09-22 22:25:25.827 [INF] TKBY: unlocked unneeded credit daf6de0381be55e4b0e1ebb12c44eef2a0c7e718ba4f9de65e57ce29ffc30ad3:2
```
Running lastest dcrd/dcrwallet master and vspd with this PR merged. Voting nodes have --manualtickets.

**Comment by jholdstock** [comment]
Looks unrelated to this PR. Seems strange for ticketbuyer to run into insufficient balance errs, maybe @dajohi or @jrick know whats up. Otherwise it could be worth opening a dcrwallet issue.

**Comment by isuldor** [comment]
okay, I'm using `purchaseaccount=default` and guessing that everyone else is testing this cose using the coinshuffle server. I'm waiting to see if the wallet will work with that and then will open a case. Otherwise I may just be doing something wrong.  Will update when the next epoch closes.

**Review by jholdstock — APPROVED** [review]
I've read through the dcrd funcs which have the new `isTreasuryEnabled` param, and afaict, hardcoding `true` should have no impact on the operation of vspd. LGTM.

**Pull request merged**